packages/@ember/-internals/runtime/lib/system/core_object.ts

```diff
@@ -666,9 +707,15 @@ class CoreObject {
     @param {Object} [arguments]* Object containing values to use within the new class
     @public
   */
-  static extend() {
+  // Args should be `Array<Mixin | Record<string, unknown>>` but this causes
+  // types to be overly strict.
+  static extend<Statics, Instance>(
```

Comment:
Just leaving a note for us that we're going to want to think about how to handle these when we start publishing types: I think (?!?) and wrote in the RFC that we *don't* want to publish these, because they mostly give people just enough rope to hang themselves with in classic classes.

Reply:
Yeah, I think that’s probably correct